pkg/cluster/validate_create.go

```diff
@@ -0,0 +1,68 @@
+package cluster
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/api/admission/v1beta1"
+	capiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-admission-controller/internal/errors"
+	"github.com/giantswarm/azure-admission-controller/pkg/validator"
+)
+
+type CreateValidator struct {
+	baseDomain string
+	ctrlClient client.Client
+	logger     micrologger.Logger
+}
+
+type CreateValidatorConfig struct {
+	BaseDomain string
+	CtrlClient client.Client
+	Logger     micrologger.Logger
+}
+
+func NewCreateValidator(config CreateValidatorConfig) (*CreateValidator, error) {
+	if config.BaseDomain == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.BaseDomain must not be empty", config)
+	}
+	if config.CtrlClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CtrlClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	v := &CreateValidator{
+		ctrlClient: config.CtrlClient,
```

Comment:
You are not setting the `baseDomain`. And it seems the client is not used.

Reply:
:( thanks